projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -228,13 +229,21 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
           formArray.removeAt(formArray.length - 1);
         }
 
-        while (formArray.length < value.length) {
-          formArray.push(this.createFormArrayControl(key as ArrayPropertyOf<FormInterface>));
+        for (let i = formArray.length; i < value.length; i++) {
+          if (this.formIsFormWithArrayControls()) {
+            formArray.insert(i, this.createFormArrayControl(key as ArrayPropertyOf<FormInterface>, value[i]));
```

Comment:
[Push](https://angular.io/api/forms/FormArray#push):  

> Insert a new AbstractControl at the end of the array.

[Insert](https://angular.io/api/forms/FormArray#insert)

> Insert a new AbstractControl at the given index in the array.

I think `push` is more appropriated in that case

Reply:
```ts
formArray.insert(i, this.createFormArrayControl(key as ArrayPropertyOf<FormInterface>, value[i]));
```

I think that the second argument here shouldn't be called `initialValue` :thinking: It's more like `currentValue` isn't it? or `valueToBeApplied` but initial value feels to me like you'd pass the default one

Reply:
I refactored to use a for loop as I needed to index the `value[i]` to pass to the `createFormArrayControl` function, given I already have the index it make much more sense in terms of resilience to use the insert style as if we later do some filtering or something there is no risk of the input array not matching the structure of the resulting formarray.

The `insert` method does still increase the size of the way - it's the equivalent of

```js
foo = [];
foo[2]  = true;
console.log(foo); // [empty, empty, true]
```